qiskit_ibm_runtime/sampler.py

```diff
@@ -169,6 +169,18 @@ def __init__(
         _options.transpilation.skip_transpilation = (  # type: ignore[union-attr]
             skip_transpilation
         )
+
+        if _options.optimization_level is None:
+            if _options.simulator and (
+                not hasattr(_options.simulator, "noise_model")
+                or asdict(_options.simulator)["noise_model"] is None
+            ):
+                _options.optimization_level = 1
+                _options.resilience_level = 0
+            else:
+                _options.optimization_level = Options._DEFAULT_OPTIMIZATION_LEVEL
+                _options.resilience_level = Options._DEFAULT_RESILIENCE_LEVEL
+
         self._options: dict = asdict(_options)
```

Comment:
Sorry, didn't see this initially - users can set optimization level or resilience level separately so the logic would have to something like: 

```python
if _options.optimization_level is None:
      # set values appropriately 
            
if _options.resilience_level is None:
      # set values appropriately 
            
```

Reply:
Of course you are right - I don't know how I missed this. Will fix.